oci/layout/oci_transport.go

```diff
@@ -63,7 +63,8 @@ type ociReference struct {
 	resolvedDir string // Absolute path with no symlinks, at least at the time of its creation. Primarily used for policy namespaces.
 	// If image=="", it means the "only image" in the index.json is used in the case it is a source
 	// for destinations, the image name annotation "image.ref.name" is not added to the index.json
-	image string
+	image       string
+	sourceIndex int
```

Comment:
This _must_ show up in `StringWithinTransport`, and the tests should have appropriate coverage.